library/std/src/sys/windows/path/absolute.rs

```diff
@@ -0,0 +1,76 @@
+// Original implementation taken from crate "normpath".
+// Copyright 2020 "dylni" (https://github.com/dylni)
+
+use super::super::{c, fill_utf16_buf, os2path, to_u16s, IsZero};
+use crate::io;
+use crate::path::{Path, PathBuf};
+use crate::ptr;
+
+#[cfg(test)]
+mod tests;
+
+/// Calls `GetFullPathNameW` with a workaround for incorrect results.
+pub(crate) unsafe fn get_full_path_name(
+    lpfilename: *const u16,
+    buffer: *mut u16,
+    size: c::DWORD,
+) -> c::DWORD {
+    // While the docs for `GetFullPathNameW` have the standard note about
+    // needing a `\\?\` path for a long `lpfilename`, this does not appear to
+    // be true in practice.
+    // See:
+    // https://stackoverflow.com/questions/38036943/getfullpathnamew-and-long-windows-file-paths
+    // https://googleprojectzero.blogspot.com/2016/02/the-definitive-guide-on-win32-to-nt.html
+    let result = c::GetFullPathNameW(lpfilename, size, buffer, ptr::null_mut());
+    if result.is_zero() {
+        return result;
+    }
+
+    // WinAPI can recommend an insufficient size that causes it to return
+    // incorrect results, so extra space is reserved as a workaround:
+    // https://github.com/dylni/normpath/issues/5
+    let required_size = result.saturating_add(2);
+    if size < required_size { required_size } else { result }
```

Comment:
`fill_utf16_buf` will use a [minimum buffer size of 512 code units](https://github.com/rust-lang/rust/blob/2184c7c5687a89ce58e169546c9b83448f4d0ea1/library/std/src/sys/windows/mod.rs#L196-L202). Given that, is this workaround still necessary?

Reply:
Nice catch. You're right that this is unnecessary. I haven't seen any larger inputs that cause an incorrect result.